homeassistant/components/binary_sensor/ping.py

```diff
@@ -0,0 +1,133 @@
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.util import Throttle
+from homeassistant.const import CONF_NAME, CONF_HOST
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ROUND_TRIP_TIME_AVG = 'round_trip_time_avg'
+ATTR_ROUND_TRIP_TIME_MAX = 'round_trip_time_max'
+ATTR_ROUND_TRIP_TIME_MDEV = 'round_trip_time_mdev'
+ATTR_ROUND_TRIP_TIME_MIN = 'round_trip_time_min'
+
+CONF_PING_COUNT = 'count'
+
+DEFAULT_NAME = 'Ping Binary sensor'
+DEFAULT_PING_COUNT = 5
+DEFAULT_SENSOR_CLASS = 'connectivity'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+PING_MATCHER = re.compile(
+    r'(?P<min>\d+.\d+)\/(?P<avg>\d+.\d+)\/(?P<max>\d+.\d+)\/(?P<mdev>\d+.\d+)')
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PING_COUNT, default=DEFAULT_PING_COUNT): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Ping Binary sensor."""
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    count = config.get(CONF_PING_COUNT)
+
+    add_devices([PingBinarySensor(name, PingData(host, count))], True)
+
+
+class PingBinarySensor(BinarySensorDevice):
+    """Representation of a Ping Binary sensor."""
+
+    def __init__(self, name, ping):
+        """Initialize the Ping Binary sensor."""
+        self._name = name
+        self.ping = ping
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return DEFAULT_SENSOR_CLASS
+
+    @property
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+        return self.ping.available
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the ICMP checo request."""
+        if self.ping.data is not False:
+            return {
+                ATTR_ROUND_TRIP_TIME_AVG: self.ping.data['avg'],
+                ATTR_ROUND_TRIP_TIME_MAX: self.ping.data['max'],
+                ATTR_ROUND_TRIP_TIME_MDEV: self.ping.data['mdev'],
+                ATTR_ROUND_TRIP_TIME_MIN: self.ping.data['min'],
+            }
+
+    def update(self):
+        """Get the latest data."""
+        self.ping.update()
+
+
+class PingData(object):
+    """The Class for handling the data retrieval."""
+
+    def __init__(self, host, count):
+        """Initialize the data object."""
+        self._ip_address = host
+        self._count = count
+        self.data = {}
+        self.available = False
+
+        if sys.platform == 'win32':
+            self._ping_cmd = [
+                'ping', '-n', str(self._count), '-w 1000', self._ip_address]
+        else:
+            self._ping_cmd = [
+                'ping', '-n', '-q', '-c', str(self._count), '-W1',
```

Comment:
This may not work for IPv6 hosts see https://github.com/home-assistant/home-assistant/issues/6161